File: KKProject_ReactiveObjC/Home/ScrollView/KKTableView.m
//
//  KKTableView.m
//  KKProject_ReactiveObjC
//
//  Created by youbin on 2020/7/6.
//  Copyright © 2020 Macbook Pro 15.4 . All rights reserved.
//

#import "KKTableView.h"

@implementation KKTableView

/// 识别多手势
/// @param gestureRecognizer gestureRecognizer
/// @param otherGestureRecognizer otherGestureRecognizer
-(BOOL)gestureRecognizer:(UIGestureRecognizer *)gestureRecognizer shouldRecognizeSimultaneouslyWithGestureRecognizer:(UIGestureRecognizer *)otherGestureRecognizer{
    return true;
}

/*
// Only override drawRect: if you perform custom drawing.
// An empty implementation adversely affects performance during animation.
- (void)drawRect:(CGRect)rect {
    // Drawing code
}
*/

@end

File: KKProject_Swift/Home/KKTableView/KKTableView.swift
//
//  KKTableView.swift
//  KKProject_Swift
//
//  Created by youbin on 2020/1/25.
//  Copyright © 2020 Macbook Pro 15.4 . All rights reserved.
//

import UIKit

class KKTableView : UIView , KKTableViewProtocol, UITableViewDelegate,UITableViewDataSource{
    
    public lazy var tableView : UITableView = {
        let view = UITableView.init(frame: CGRect.zero, style: .plain)
        view.delegate = self
        view.dataSource = self
        view.showsHorizontalScrollIndicator = false
        view.separatorStyle = .none
        return view
    }()
    
    var viewModel : KKViewModelProtocol?
    
    init(viewmodel : KKViewModelProtocol) {
        super.init(frame: CGRect.zero)
        viewModel = viewmodel
        kk_setupView()
        kk_bindViewModel()
    }
    
    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    func kk_setupView() {
        self.addSubview(tableView)
        tableView.snp.makeConstraints { (make) in
            make.edges.equalToSuperview()
        }
    }
    
    func kk_bindViewModel() {
        
    }
    
    /*
     // Only override draw() if you perform custom drawing.
     // An empty implementation adversely affects performance during animation.
     override func draw(_ rect: CGRect) {
     // Drawing code
     }
     */
    
}

extension KKTableView{
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return 0
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        return UITableViewCell()
    }
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        tableView.deselectRow(at: indexPath, animated: true)
    }
    
    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
        return 0.000001
    }
    
    func tableView(_ tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
        return 0.000001
    }
    
    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
         return 0.000001
    }
}
